### PR TITLE
Add ToDecibels CPU operator

### PR DESCRIPTION
**Pull request by jantonguirao**
#### Why we need this PR?
-  Some magnitudes are often represented in dB. We need an operator for that

#### What happened in this PR?
-  Implemented ToDecibels operator for the CPU
-  Added Kernel unit tests

**JIRA TASK**: [DALI-1173]

### CODE REVIEW

File: dali/operators/signal/decibel/to_decibels.cc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      R"code(Factor by which we multiply the logarithm (typically 10.0 or 20.0 depending if we
+are dealing with a squared magnitude or not).)code",
+      10.0f)
+    .AddOptionalArg("reference",
+      R"code(Reference magnitude. If not provided, the maximum of the input will be used as
+reference. Note: The maximum of the input will be calculated on a per-sample basis.)code",
+      1.0f)
+    .AddOptionalArg("min_ratio",
+      R"code(Minimum or cut-off ratio. Any ratio below this value will saturate. Example:
+A value `min_ratio=1e-8` corresponds to a dB cut-off of -80 dBs (assuming `multiplier=10.0`).)code",
```
Comment (JanuszL) [inline dali/operators/signal/decibel/to_decibels.cc]: Maybe min_ratio should be provided in dBs not as a decimal value?

File: dali/operators/signal/decibel/to_decibels.cc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  auto &output = ws.OutputRef<CPUBackend>(0);
+  kernels::KernelContext ctx;
+  auto in_shape = input.shape();
+  int nsamples = input.size();
+  auto nthreads = ws.GetThreadPool().size();
+
+  TYPE_SWITCH(input.type().id(), type2id, T, TO_DB_SUPPORTED_TYPES, (
+    VALUE_SWITCH(in_shape.sample_dim(), Dims, TO_DB_SUPPORTED_NDIMS, (
+      using ToDbKernel = kernels::signal::ToDecibelsCpu<T, Dims>;
+      kmgr_.Initialize<ToDbKernel>();
```
Comment (JanuszL) [inline dali/operators/signal/decibel/to_decibels.cc]: It seems as a common pattern for pass through operators - where input and output matches. Maybe we can extract it to some common place.

File: dali/operators/signal/decibel/to_decibels.cc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+template <>
+void ToDecibels<CPUBackend>::RunImpl(workspace_t<CPUBackend> &ws) {
+  const auto &input = ws.InputRef<CPUBackend>(0);
+  auto &output = ws.OutputRef<CPUBackend>(0);
+  auto in_shape = input.shape();
+  int nsamples = input.size();
+  auto& thread_pool = ws.GetThreadPool();
+
+  TYPE_SWITCH(input.type().id(), type2id, T, TO_DB_SUPPORTED_TYPES, (
```
Comment (JanuszL) [inline dali/operators/signal/decibel/to_decibels.cc]: I wonder how frequent patter is that one too.

**Comment by jantonguirao** [comment]
!build

File: dali/operators/signal/decibel/to_decibels.cc (hunk 1 above)
Comment (jantonguirao) [reply dali/operators/signal/decibel/to_decibels.cc]: Will do

File: dali/operators/signal/decibel/to_decibels.cc (hunk 2 above)
Comment (jantonguirao) [reply dali/operators/signal/decibel/to_decibels.cc]: As we agreed, it can't be easily generalized to the point of being reusable

**Comment by jantonguirao** [comment]
!build

**Pull request merged**